sktime/transformations/series/detrend/_deseasonalize.py

```diff
@@ -471,36 +467,46 @@ def fit(self, Z, X=None):
             trend_jump=self.trend_jump,
             low_pass_jump=self.low_pass_jump,
         ).fit()
+        self.seasonal_ = pd.Series(self.stl_.seasonal, index=X.index)
+        self.trend_ = pd.Series(self.stl_.trend, index=X.index)
 
-        self.seasonal_ = pd.Series(self._stl.seasonal, index=Z.index)
-        self.resid_ = pd.Series(self._stl.resid, index=Z.index)
```

Comment:
why did you remove `resid_` attribute?

Reply:
it's being returned as the columns of the result now

Reply:
I am still not a fan of returning all components at same time, what is the use case for this? User could just use the `statsmodels` implementation. We need to make it compliant with pipelining, that is more the use case of the `sktime` transformer in my opinion